src/project-converter.ts

```diff
@@ -37,154 +38,98 @@ const excludeFromResults = new Set([
  * For best results, only one ProjectConverter instance should be needed, so
  * that it can cache results and avoid duplicate, extraneous document
  * conversions.
- *
- * ProjectConverter is indifferent to the layout of the project, delegating any
- * special URL handling/resolution to the urlHandler provided to the
- * constructor.
  */
 export class ProjectConverter {
-  readonly urlHandler: UrlHandler;
-  readonly conversionSettings: ConversionSettings;
-
-  /**
-   * A cache of all JS Exports by namespace, to map implicit HTML imports to
-   * explicit named JS imports.
-   */
-  readonly namespacedExports = new Map<string, JsExport>();
+  private readonly analysis: Analysis;
+  private readonly urlHandler: UrlHandler;
+  private readonly conversionSettings: ConversionSettings;
+  private readonly scanner: ProjectScanner;
 
   /**
    * A cache of all converted documents. Document conversions should be
    * idempotent, so conversion results can be safely cached.
    */
-  readonly conversionResults = new Map<OriginalDocumentUrl, ConversionResult>();
+  private readonly results = new Map<OriginalDocumentUrl, ConversionResult>();
 
-  constructor(urlHandler: UrlHandler, conversionSettings: ConversionSettings) {
+  constructor(
+      analysis: Analysis, urlHandler: UrlHandler,
+      conversionSettings: ConversionSettings) {
+    this.analysis = analysis;
     this.urlHandler = urlHandler;
     this.conversionSettings = conversionSettings;
+    this.scanner = new ProjectScanner(analysis, urlHandler, conversionSettings);
   }
 
   /**
    * Convert a document and any of its dependencies. The output format (JS
    * Module or HTML Document) is determined by whether the file is included in
    * conversionSettings.includes.
    */
+  convertPackage(matchPackageName: string) {
+    // First, scan the package (and any of its dependencies)
+    this.scanner.scanPackage(matchPackageName);
+    // Then, convert each document in the given package
+    [...this.analysis.getFeatures({kind: 'html-document'})]
```

Comment:
Most of this `getFeatures` ... `filter` business is duplicated in `scanPackage`.  Maybe there's a way to DRY up the document query as `getHtmlDocumentsForPackage` and then call into it from `convertPackage` `scanPackage`?

Reply:
Great idea!